hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/index/bucket/HoodieSparkConsistentBucketIndex.java

```diff
@@ -275,4 +279,62 @@ public Option<HoodieRecordLocation> getRecordLocation(HoodieKey key) {
       throw new HoodieIndexException("Failed to getBucket as hashing node has no file group");
     }
   }
+
+  /**
+   * Update default metadata file(00000000000000.hashing_meta) with the latest committed metadata file so that default file will be in sync
+   * with latest commit.
+   *
+   * @param table
+   */
+  public void updateMetadata(HoodieTable table) {
+    Map<String, Boolean> partitionVisiteddMap = new HashMap<>();
+    Option<HoodieInstant> hoodieOldestReplaceInstantToKeep = getOldestInstantToRetain(table);
```

Comment:
Could the oldest instant to retain get from interface parameter? Otherwise the oldest instant gets twice and consumes some performance here.

Reply:
Yeah i can make the change in interface. let me do it

Reply:
Removed redundant code. modified interface